src/logging/Tag.h

```diff
@@ -3,13 +3,17 @@
 #pragma once
 
 #include "zeek-config.h"
-#include "util.h"
 #include "../Tag.h"
-#include "plugin/TaggedComponent.h"
-#include "plugin/ComponentManager.h"
 
 class EnumVal;
 
+namespace plugin {
+template <class T>
+class TaggedComponent;
+template <class T, class C>
+class ComponentManager;
+}
+
```

Comment:
Redefining this block in multiple places (see also `input/Tag.h`) feels a little brittle, not that it's likely to ever change.

Reply:
Each header which needs a certain symbol should have a copy of the forward declaration, even if another header has already done it. Imagine that other header one day doesn't need the declaration anymore, or that header isn't included anymore - then `logging/Tag.h` must not break, and should not need to be adjusted. It should be self-contained.
Having multiple forward declarations for one symbol is allowed in C++.